config/crd/bases/workload.nephio.org_amfdeployments.yaml

```diff
@@ -0,0 +1,366 @@
+---
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.9.2
+  creationTimestamp: null
+  name: amfdeployments.workload.nephio.org
+spec:
+  group: workload.nephio.org
+  names:
+    kind: AMFDeployment
+    listKind: AMFDeploymentList
+    plural: amfdeployments
+    singular: amfdeployment
+  scope: Namespaced
+  versions:
+  - name: v1alpha1
+    schema:
+      openAPIV3Schema:
+        properties:
+          apiVersion:
+            description: 'APIVersion defines the versioned schema of this representation
+              of an object. Servers should convert recognized schemas to the latest
+              internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
+            type: string
+          kind:
+            description: 'Kind is a string value representing the REST resource this
+              object represents. Servers may infer this from the endpoint the client
+              submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
+            type: string
+          metadata:
+            type: object
+          spec:
+            properties:
+              capacity:
+                description: capacity defines the capacity characteristics of the
+                  NF deployment
+                properties:
+                  maxDownlinkThroughput:
```

Comment:
@henderiw Do you think AMF needs throughput information ? I don't see in [free5GC](https://github.com/Orange-OpenSource/towards5gs-helm/blob/main/charts/free5gc/charts/free5gc-amf/values.yaml#L149) and in general I don't think AMF needs that. Probably we can remove this field.

Reply:
@arora-sagar  ---- what @henderiw does is to have a consolidated Capacity definition that can be used for "all" NFs (even beyond 5GC). So for now, we basically combine all three's (AMF, SMF, UPF) capacity needs into one structure. For each NF's controller, the irrelevant field can just be ignored.

Reply:
as @s3wong said we don't want to specialise to much, not all parameters are required, so you can omit when empty